src/status_im2/subs/wallet/wallet.cljs

```diff
@@ -131,31 +134,32 @@
    (remove #(:watch-only? %) accounts)))
 
 (defn- calc-token-value
-  [{:keys [market-values-per-currency] :as item} chain-id]
-  (let [crypto-value                      (utils/token-value-in-chain item chain-id)
-        market-values                     (:usd market-values-per-currency)
-        {:keys [price change-pct-24hour]} market-values
-        fiat-change                       (utils/calculate-fiat-change crypto-value change-pct-24hour)]
-    (when (and crypto-value (seq (:name item)))
-      {:token               (keyword (string/lower-case (:symbol item)))
-       :token-name          (:name item)
-       :state               :default
-       :status              (cond
-                              (pos? change-pct-24hour) :positive
-                              (neg? change-pct-24hour) :negative
-                              :else                    :empty)
-       :customization-color :blue
-       :values              {:crypto-value      crypto-value
-                             :fiat-value        (utils/prettify-balance (* crypto-value price))
-                             :percentage-change (.toFixed change-pct-24hour 2)
-                             :fiat-change       (utils/prettify-balance fiat-change)}})))
+  [{:keys [market-values-per-currency] :as token} color currency currency-symbol]
+  (let [token-units                 (utils/total-token-units-in-all-chains token)
+        fiat-value                  (utils/total-token-fiat-value currency token)
+        market-values               (get market-values-per-currency
+                                         currency
+                                         (get market-values-per-currency
+                                              constants/profile-default-currency))
+        {:keys [change-pct-24hour]} market-values]
+    {:token               (:symbol token)
+     :token-name          (:name token)
+     :state               :default
+     :status              (cond
+                            (pos? change-pct-24hour) :positive
+                            (neg? change-pct-24hour) :negative
+                            :else                    :empty)
+     :customization-color color
+     :values              {:crypto-value token-units
+                           :fiat-value   (utils/prettify-balance currency-symbol fiat-value)}}))
 
 (rf/reg-sub
  :wallet/account-token-values
  :<- [:wallet/current-viewing-account]
- :<- [:chain-id]
- (fn [[current-account chain-id]]
-   (mapv #(calc-token-value % chain-id) (:tokens current-account))))
+ :<- [:profile/currency]
+ :<- [:profile/currency-symbol]
+ (fn [[{:keys [tokens color]} currency currency-symbol]]
+   (mapv #(calc-token-value % color currency currency-symbol) tokens)))
```

Comment:
@smohamedjavid is this taking into account the different network configuration of a given account? 🤔

Reply:
@J-Son89 - Not at this moment. Will work on that separately. 👍